study/src/test/java/aop/stage0/TransactionHandler.java

```diff
@@ -1,15 +1,41 @@
 package aop.stage0;
 
+import aop.DataAccessException;
+import aop.Transactional;
 import java.lang.reflect.InvocationHandler;
 import java.lang.reflect.Method;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.DefaultTransactionDefinition;
 
 public class TransactionHandler implements InvocationHandler {
 
+    private final PlatformTransactionManager platformTransactionManager;
+    private final Object target;
+
+    public TransactionHandler(PlatformTransactionManager platformTransactionManager, Object target) {
+        this.platformTransactionManager = platformTransactionManager;
+        this.target = target;
+    }
+
     /**
      * @Transactional 어노테이션이 존재하는 메서드만 트랜잭션 기능을 적용하도록 만들어보자.
      */
     @Override
     public Object invoke(final Object proxy, final Method method, final Object[] args) throws Throwable {
-        return null;
+        if (!target.getClass().getMethod(method.getName(), method.getParameterTypes())
+                .isAnnotationPresent(Transactional.class)) {
```

Comment:
```suggestion
        if (!method.isAnnotationPresent(Transactional.class)) {
```

Transactional 어노테이션이 없는 경우를 판단하는 로직 같아요!
Method 클래스의 isAnnotationPresent를 사용하면 가독성이 좋아질 것 같은데 어떤가요?ㅎㅎ

Reply:
실제 구현체(target)가 가지고 있는 method에서 어노테이션 검증을 하는 방식이 아닌, 파라미터로 받은 method를 통해 검증하게 되면 interface의 메서드를 검증하게 되어 `@Transactional`이 없다는 결과가 출력돼요! 따라서 위와 같이 target 구현체의 method를 사용하게 되었습니다 :)